src/coordinate_search.jl

```diff
@@ -0,0 +1,65 @@
+export coordinate_search
+
+"""
+	The coordinate search is a derivative free optimization method.
+	It minimizes a certain function f by walking on a grid and taking a step where the function values are smaller.
+	When it can no longer find any smaller values, it reduces the grid by a factor of 2, repeating the process.
+
+"""
+
+function coordinate_search(nlp :: AbstractNLPModel;
+			   tol = 1e-12,
+			   k = 0,
+			   α = 1.0,
+			   xk = nlp.meta.x0,
+			   el_time = 0.0,
+			   max_time :: Real  = 1e-1,
+			   max_eval = 500)
+
+
+start_time = time()
+tired = neval_obj(nlp) > max_eval || el_time > max_time
+optimal = α <= tol
+T = eltype(xk)
+status =:unknown
+@info log_header([:iter, :f, :H],[Int, T, T], hdr_override=Dict(:f=>"f(x)", :H=>"α"))
+
+while !(optimal || tired)
+	f = obj(nlp,xk)
```

Comment:
Store `f` at the current iteration, and reuse it. Also happens at lines 36 and 63.